tests/returntypes/test_finite_difference_shot_vec_new.py

```diff
@@ -0,0 +1,1116 @@
+            h=10e-2,
+            shots=many_shots_shot_vector,
+        )
+        all_res = fn(dev.batch_execute(tapes))
+
+        assert isinstance(all_res, tuple)
+        assert len(all_res) == len(default_shot_vector)
+
+        for res in all_res:
+
+            assert isinstance(res, tuple)
+            assert len(res) == 2
+
+            assert isinstance(res[0], tuple)
+            assert len(res[0]) == 2
+            assert np.allclose(res[0][0], -np.sin(x), atol=finite_diff_shot_vec_tol, rtol=0)
+            assert isinstance(res[0][0], numpy.ndarray)
+            assert np.allclose(res[0][1], 0, atol=finite_diff_shot_vec_tol, rtol=0)
+            assert isinstance(res[0][1], numpy.ndarray)
+
+            assert isinstance(res[1], tuple)
+            assert len(res[1]) == 2
+            assert np.allclose(
+                res[1][0],
+                [
+                    -(np.cos(y / 2) ** 2 * np.sin(x)) / 2,
+                    -(np.sin(x) * np.sin(y / 2) ** 2) / 2,
+                    (np.sin(x) * np.sin(y / 2) ** 2) / 2,
+                    (np.cos(y / 2) ** 2 * np.sin(x)) / 2,
+                ],
+                atol=finite_diff_shot_vec_tol,
+                rtol=0,
+            )
+            assert isinstance(res[1][0], numpy.ndarray)
+            assert np.allclose(
+                res[1][1],
+                [
+                    -(np.cos(x / 2) ** 2 * np.sin(y)) / 2,
+                    (np.cos(x / 2) ** 2 * np.sin(y)) / 2,
+                    (np.sin(x / 2) ** 2 * np.sin(y)) / 2,
+                    -(np.sin(x / 2) ** 2 * np.sin(y)) / 2,
+                ],
+                atol=finite_diff_shot_vec_tol,
+                rtol=0,
+            )
+            assert isinstance(res[1][1], numpy.ndarray)
+
+
+@pytest.mark.parametrize("approx_order", [2])
+@pytest.mark.parametrize("strategy", ["center"])
+@pytest.mark.xfail(reason="TODO: higher-order derivatives with finite shots")
+class TestFiniteDiffGradients:
+    """Test that the transform is differentiable"""
+
+    @pytest.mark.autograd
+    def test_autograd(self, approx_order, strategy, tol):
+        """Tests that the output of the finite-difference transform
+        can be differentiated using autograd, yielding second derivatives."""
+        dev = qml.device("default.qubit.autograd", wires=2, shots=many_shots_shot_vector)
+        params = np.array([0.543, -0.654], requires_grad=True)
+
+        def cost_fn(x):
+            with qml.tape.QuantumTape() as tape:
+                qml.RX(x[0], wires=[0])
+                qml.RY(x[1], wires=[1])
+                qml.CNOT(wires=[0, 1])
+                qml.expval(qml.PauliZ(0) @ qml.PauliX(1))
+
+            tape.trainable_params = {0, 1}
+            tapes, fn = finite_diff(
+                tape,
+                n=1,
+                approx_order=approx_order,
+                strategy=strategy,
+                h=10e-2,
+                # shots=many_shots_shot_vector,
```

Comment:
Is it done on purposee?

Reply:
It's xfail, but will remove the comment.